src/gei/Commands/MigrateRepoCommand.cs

```diff
@@ -100,6 +103,18 @@ public class MigrateRepoCommand : CommandBase<MigrateRepoCommandArgs, MigrateRep
         {
             Description = "If uploading to S3, the AWS secret key. If not provided, it will be read from AWS_SECRET_KEY environment variable."
         };
+        public Option<string> AwsSessionToken { get; } = new("--aws-session-token")
+        {
+            Description = "If uploading to S3, the AWS session token."
+        };
+        public Option<string> AwsRegion { get; } = new("--aws-region")
+        {
+            Description = "If uploading to S3, the AWS resion, Default valule is us-east-1"
+        };
+        public Option<string> AwsS3UseSignatureVersion4 { get; } = new("--aws-s3-useSignatureVersion4")
```

Comment:
I don't have a tonne of context on S3, so this might be a stupid question, but is there any reason why we shouldn't _always_ use V4? Are there cases where someone would want to use V3?

From what I have seen in my own research, some S3 regions don't support V3. Perhaps there are some that don't support V4!

Reply:
The AWS docs seem to strongly push V4 as the right option nowadays: https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingAWSSDK.html

If that is the case, I think we should seriously consider switching to V4 and not having an option: https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingAWSSDK.html#UsingAWSSDK-sig2-deprecation

Reply:
I'd been trying to get a migration to work most of today, and (thanks to this pull request and issue #744) I finally figured out the issue was because I was using a bucket in region eu-west-2.

Assuming v4 works in all regions, I would strongly support the proposal to always use the v4 algorithm, so other users don't encounter failures for mysterious reasons.

Allowing users to chose region and algorithm might still leave them susceptible to mysterious failure, especially if the defaults didn't work for their region.

If it is decided to leave it as a user option, I'd suggest that it is at least defaulted to v4, to ensure a user would have to change explicitly change it to select an algorithm that might not work for them.

Though, I can see no reason someone would select the older algorithm, so I would keep things simple, and always use v4.

Reply:
I had precisely the same experience when using `eu-west-1` - so you're not alone 😓